xarray/coding/cftimeindex.py

```diff
@@ -541,14 +541,29 @@ def shift(self, n, freq):
         >>> index.shift(1, "M")
         CFTimeIndex([2000-02-29 00:00:00],
                     dtype='object', length=1, calendar='gregorian', freq=None)
+        >>> index.shift(1.5, "D")
+        CFTimeIndex([2000-02-01 12:00:00],
+                    dtype='object', length=1, calendar='gregorian', freq=None)
         """
         from .cftime_offsets import to_offset
 
-        if not isinstance(n, int):
-            raise TypeError(f"'n' must be an int, got {n}.")
+        if not isinstance(n, (int, float)):
+            raise TypeError(f"'n' must be an int or float, got {n}.")
```

Comment:
I think you can remove this check. Add some typing instead, `n: int | float, str: frq | timedelta`.